Gems/Atom/Feature/Common/Assets/Shaders/Materials/BasePBR/BasePBR_SurfaceEval.azsli

```diff
@@ -52,6 +53,10 @@ Surface EvaluateSurface_BasePBR(
     float2 baseColorUv = uvs[MaterialSrg::m_baseColorMapUvIndex];
     real3 sampledColor = GetBaseColorInput(MaterialSrg::m_baseColorMap, MaterialSrg::m_sampler, baseColorUv, real3(MaterialSrg::m_baseColor.rgb), o_baseColor_useTexture, uvDxDy, customDerivatives);
     real3 baseColor = BlendBaseColor(sampledColor, real3(MaterialSrg::m_baseColor.rgb), real(MaterialSrg::m_baseColorFactor), o_baseColorTextureBlendMode, o_baseColor_useTexture);
+    if(o_useVertexColor && o_color0_isBound)
```

Comment:
why have two separate shader options? Cant we get away with just one?

Reply:
Ahh I see. It wasn't clear to me. Maybe nice to add a comment where o_color0_isBound is defined that it is meant to indicate if a mesh has color vertex stream.